src/app/components/wrapper-components/online-game-wrapper/online-game-wrapper.component.ts

```diff
@@ -216,357 +213,225 @@ export class OnlineGameWrapperComponent extends GameWrapper<MinimalUser> impleme
+        this.endGame = true;
+        this.timeManager.onGameEnd();
     }
-    public getUpdateTypes(update: PartDocument): UpdateType[] {
-        const currentPartDoc: Part | null = this.currentPart != null ? this.currentPart.data : null;
-        const diff: ObjectDifference = ObjectDifference.from(currentPartDoc, update.data);
-        const updatesTypes: UpdateType[] = [];
-        display(OnlineGameWrapperComponent.VERBOSE, { currentPartDoc, update, diff });
-        const nbDiffs: number = diff.countChanges();
-        if (nbDiffs === 0) {
-            updatesTypes.push(UpdateType.DUPLICATE);
-        }
-        if (diff.isFullyCreated() && diff.isPresent('beginning').present) {
-            updatesTypes.push(UpdateType.STARTING_DOC);
-        }
-        if (update.data.request) {
-            const lastMoveTimeIsRemoved: boolean = diff.removed.lastUpdateTime != null;
-            if (update.data.request.code === 'TakeBackAccepted' && lastMoveTimeIsRemoved) {
-                updatesTypes.push(UpdateType.ACCEPT_TAKE_BACK_WITHOUT_TIME);
-            } else {
-                updatesTypes.push(UpdateType.REQUEST);
-            }
-        }
-        if (currentPartDoc == null) {
-            if (diff.isPresent('listMoves').state === 'added' && update.data.turn > 0) {
-                updatesTypes.push(this.getMoveUpdateType(update, diff));
-            }
-        } else if (this.turnHasIncreased(currentPartDoc.turn, update.data.turn, update.data.listMoves)) {
-            updatesTypes.push(this.getMoveUpdateType(update, diff));
-        }
-        if (update.data.beginning == null) {
-            updatesTypes.push(UpdateType.PRE_START_DOC);
-        }
-        if (diff.isPresent('result').present &&
-            update.data.result !== MGPResult.UNACHIEVED.value)
-        {
-            const turnModified: boolean = diff.modified.turn != null;
-            const lastUpdateTimeMissing: boolean = diff.addedOrModified('lastUpdateTime') === false;
-            if (turnModified && lastUpdateTimeMissing) {
-                updatesTypes.push(UpdateType.END_GAME_WITHOUT_TIME);
-            } else {
-                updatesTypes.push(UpdateType.END_GAME);
-            }
-        }
-        return updatesTypes;
+    private async onReceivedMove(moveEvent: PartEventMove): Promise<void> {
+        const rules: Rules<Move, GameState, unknown> = this.gameComponent.rules;
+        const currentPartTurn: number = this.gameComponent.getTurn();
+        const chosenMove: Move = this.gameComponent.encoder.decode(moveEvent.move);
+        const legality: MGPFallible<unknown> = rules.isLegal(chosenMove, this.gameComponent.getState());
+        const message: string = 'We received an incorrect db move: ' + chosenMove.toString() +
+            ' at turn ' + currentPartTurn +
+            'because "' + legality.getReasonOr('') + '"';
+        assert(legality.isSuccess(), message);
+        const success: boolean = rules.choose(chosenMove);
```

Comment:
heeeey tueur de code ! J'avais rendu les logs de cet endroit plus utiles et t'as tout jartééééé ! Remet ça, didju !

Reply:
j'ai rajouté chosenMove à printer, mais il n'y a plus ce fameux "listMoves" à printer ! C'est pas une liste de mouvement qu'est illégale mais bien un mouvement, donc c'est ça qu'est intéressant à garder en plus non ?

Reply:
non du coup pas besoin de le console.logger en plus in fine !